### DISCUSSION TITLE
Custom fonts

### DISCUSSION THREAD

**Discussion by montex12** [AvaloniaUI/Avalonia#11822] [Q&A] [answered]
Hello,

I know there are many issues about font handling in Avalonia but there's not much documentation about how to load multiple fonts.

I want to use Noto Sans (with all variants, Noto Sans KR, Noto Sans SC, Noto Sans JP, etc...) but not sure how to handle that as they are separate font files.

I took a look at Avalonia.Fonts.Inter and how default Avalonia themes use them:
`<FontFamily x:Key="ContentControlThemeFontFamily">fonts:Inter#Inter, $Default</FontFamily>`

As I understand from above line, it uses regular Inter and if glyphs are missing, they fallback to $Default? (Default meaning system fonts?)

The case is that my application can show multiple languages and I want to tell Avalonia to use the proper font family based on text characters.

I have a TextBlock that can have:
Hello World --> Noto Sans
헬로 월드    --> Noto Sans KR
你好世界     --> Noto Sans SC

How can I set globally the fonts?

I tried:
`<FontFamily x:Key="ContentControlThemeFontFamily">fonts:Noto#Noto Sans, fonts:Noto#Noto Sans KR, fonts:Noto#Noto Sans JP, fonts:Noto#Noto Sans SC, $Default</FontFamily>`

But seems it's not working correctly...

I am with the latest Avalonia 11.0.0 RC1.1

**Comment by Gillibald** [comment, marked as answer]
You can use FontManagerOptions to specify global fallbacks. Just register the options in the AppBuilder.

**Comment by montex12** [comment]
Thanks!
With fallback options there's a way to set also the unicode ranges.

**Reply by CTUzkc** [reply]
Can you provide some detail codes for With fallback options ?
I also want to solve the problem with this Chinese font，because my app threw exception "default font family name can not be null or empty" on ubuntu 16.04 x64.
